pkg/kubelet/kubelet.go

```diff
@@ -1151,6 +1156,23 @@ type Kubelet struct {
 	keepTerminatedPodVolumes bool // DEPRECATED
 }
 
+// validateKubeletRuntimeConfigurationCompatibility verifies that Kubelet and runtime configurations are compatible
+func (kl *Kubelet) validateKubeletRuntimeConfigurationCompatibility(runtime kubecontainer.Runtime) error {
+	// TODO(vikasc): validate cgroup driver configuration also in this function.
+	ci := runtime.GetRuntimeConfigInfo()
+	if ci == nil {
+		return fmt.Errorf("container runtime info is empty")
+	}
+	glog.V(4).Infof("Container runtime config info: %v", ci)
+
+	// First verify that kubelet's user-namespace configuration is consistent with runtime's
+	// userns configuration
+	if kl.nodeUserNamespace != ci.IsUserNamespaceEnabled() {
+		return fmt.Errorf("container runtime and Kubelet configuration mismatch. Either both should enable or disable user-namespaces")
```

Comment:
this reads awkwardly.  it would be good to know if the kubelet has it enabled and the runtime does not or vice-versa.